### PR TITLE
Added Social Media Icons Widget

### LINKED ISSUE #2019: Widgets: add in the Social Icons widget from WordPress.com
See https://en.blog.wordpress.com/2015/04/17/social-media-icons-widget/

### PR DESCRIPTION
**Pull request by jessefriedman**
Added the Social Media Icons Widget that exists on dotcom and was authored by Chris Rudzki

fixes #2019

### CODE REVIEW

File: modules/widgets/social-media-icons.php [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+		if ( ! empty( $instance['vimeo_username'] ) ) {
+			$html .= '<li><a title="' . sprintf( $alt_text, esc_attr( $instance['vimeo_username'] ), 'Vimeo' ) . '" href="' . esc_url( 'https://vimeo.com/' . $instance['vimeo_username'] . '/' ) . '" class="genericon genericon-vimeo" target="_blank"><span class="screen-reader-text">' . sprintf( $alt_text, esc_html( $instance['vimeo_username'] ), 'Vimeo' ) . '</span></a></li>';
+		}
+
+		$html .= '</ul>';
+
+		// after widget arguments
+		$html .= $args['after_widget'];
+
+		echo apply_filters( 'wpcom-social-media-icons-widget-output', $html );
```
Comment (jeherve) [inline modules/widgets/social-media-icons.php]: There is a missing docblock here.

**Comment by jeherve** [comment]
Since we're adding a new widget, we'll need to add a mention of it [here](https://github.com/Automattic/jetpack/blob/730fcc62d7aeda5f6eeb852079e98dbb18b44006/modules/module-info.php#L398).

**Comment by jeherve** [comment]
Tested, and made some changes to fix errors. It seems to work in my tests now.

**Comment by dereksmart** [comment]
We need to keep in mind this needs to stay in sync with dotcom.  I may handle this by making some of these changes over there, and then adding the widget here via the build script so it will start with a common ancestor.

**Comment by dereksmart** [comment]
this was merged in #2230

**Pull request closed without merging**